src/statement.d

```diff
@@ -5149,6 +5154,19 @@ public:
                 }
             }
         }
+
+        if (sc.func)
+        {
+            flags |= sc.func.flags & (FUNCFLAGcppExceptions | FUNCFLAGdExceptions);
+            sc.func.flags |= flags;
+            if (flags == (FUNCFLAGcppExceptions | FUNCFLAGdExceptions))
+            {
+                error("cannot mix catching D and C++ exceptions in the same function");
```

Comment:
I'm afraid the Ada example says nothing whatsoever about what is going on under the hood, and so is not enlightening.